src/models/subscription.ts

```diff
@@ -212,10 +211,25 @@ export default class Subscription extends Sequelize.Model {
 
 	// This is a workaround to fix a long standing bug in sequelize for JSON data types
 	// https://github.com/sequelize/sequelize/issues/4387
+	setSyncState(updatedState: RepoSyncState): Subscription {
+		this.setRepoSyncState(updatedState, ["repoSyncState"]);
+		return this;
+	}
+
 	async updateSyncState(updatedState: RepoSyncState): Promise<Subscription> {
-		this.repoSyncState = _.merge(this.repoSyncState, updatedState);
-		this.changed("repoSyncState", true);
-		return this.save();
+		this.setSyncState(updatedState);
+		await this.save();
+		return this;
+	}
+
+	private setRepoSyncState(obj: Partial<Record<keyof RepoSyncState, unknown>>, parentKeys: string[], options?: SetOptions): void {
+		Object.entries(obj).forEach(([key, value]) => {
+			if (_.isPlainObject(value) && !_.isEmpty(value)) {
+				this.setRepoSyncState(value as Record<string, unknown>, parentKeys.concat([key]), options);
+			} else {
+				this.set(parentKeys.join(".") as any, value, options);
+			}
+		});
```

Comment:
So this method recursively set values to the sync state jsonb object? So we are doing it to avoid it being parsed by postgres?

Reply:
yeah, something in sequelize not liking setting a JSON object to be saved, but if you do it for each individual field in the JSON like `repoSyncState.repos.foo` with value of `bar`, it seems to work alright.  I think it's how it generate the SQL which is different and let postgres handle the individual field value change :)